specs/AppLifecycle/Restart/restartApi.markdown

```diff
@@ -0,0 +1,259 @@
+<td><p>Equivalent to CoreApplication.RequestRestartAsync, except that it is <em>not</em> async.
+<ul>
+<li>If the restart fails, but the user subsequently launches the app manually, the app will launch normally and no restart arguments will be passed. </li>
+<li>If the app has any in-process background tasks running when it calls this API, those tasks will be cancelled in the normal way. Out-of-process background tasks will not be affected. </li>
+</ul>
+ The restarted instance will be activated with Windows.ApplicationModel.Activation.LaunchActivatedEventArgs.</p>
+ <ul>
+<li>When the app is restarted, LaunchActivatedEventArgs.PreviousExecutionState will have the value Terminated so that the app can distinguish between a resume and a restart. </li>
+</ul>
+<p>If the request to restart is successful, the application will be restarted immediately. The user does not have to do anything else/handle a return value in any way.
+
+If the request fails due to a nonfatal reason, a failure reason is returned for why the restart didn’t successfully complete. This method leverages an existing enum (AppRestartFailureReason) which has the following relevant return fields.
+<ul>
+<li> InvalidUser: Could not restart for the specified user.</li>
+<li> RestartPending: A restart is already in progress.</li>
+<li> Other: Unspecified failure.</li>
+</ul>
+</p></td>
+</tr>
+</tbody>
+</table>
+
+When an application calls RequestRestartNow, the mechanism for restart will be as follows:
+1. Application calls RequestRestartNow API.
+2. The API will CreateProcess with the Agent (a helper EXE in the framework package containing the API) and will then wait for an an acknowledgment (ACK) from the Agent.
+3. The Agent will get the executable path of the application and signal an ACK to the API. The Agent will then then terminate the application and then CreateProcess with the application (restarted).
+
+## Examples and Scenarios
+
+```c#
+public class App 
+{
+    // Let's assume this method is run when an app updates assets.
+    // The new assets have downloaded and installed, and now the app has to request a restart immediately
+    protected void UpdateInstallComplete()
+    {
+        // Checking if the assets have updated successfully and there are no pending updates
+        if (!CheckForUpdate())
+        {
+            AppRestartFailureReason result = AppLifecycle::RequestRestartNow("args for new instance");
+            switch (result)
+            {
+                case AppRestartFailureReason::RestartPending:
+                    // In this case the restart could not be completed because another restart is pending and we can try to restart again after a period of time
+                    RetryRestartAfterXSeconds();
+                    break;
+                case AppRestartFailureReason::Other:
+                    // In this case, a dialog box will appear informing of an error requesting restart
```

Comment:
Can the sample show how the dialog box would get the correct message text to show the end user?

Reply:
SAMPLE: It will likely say "something went wrong".  There's not much to say other than "you will need to manually fix it, user."

Reply:
Updated